### PR TITLE
ExiledWith: fix clone interaction

### LINKED ISSUE #1122: Problem with ExiledWith Remember and Token
* Toluz, Clever Conductor on the field, it exiled some cards i discarded.
* make a token copy of it
* the token dies because of legend rule
* i somehow get the exiled cards back, but i shouldn't

### PR DESCRIPTION
**Pull request by tool4ever**
Closes #1122

I've considered switching to only `getOriginalAbility()` for compare but this should be ok until the big rework can be done.

I've added a TODO in ``cleanupExiledWith()`` with another (already existing) area of concern but since this thing is rather complex I'd rather take baby steps for improving.

### CODE REVIEW

File: forge-gui/res/cardsfolder/p/profane_procession_tomb_of_the_dusk_rose.txt [hunk 1]
```diff
@@ -1,10 +1,8 @@
 Name:Profane Procession
 ManaCost:1 W B
 Types:Legendary Enchantment
-A:AB$ ChangeZone | Cost$ 3 W B | ValidTgts$ Creature | TgtPrompt$ Choose target creature. | Origin$ Battlefield | Destination$ Exile | RememberTargets$ True | SubAbility$ DBTransform | SpellDescription$ Exile target creature. Then if there are three or more cards exiled with CARDNAME, transform it.
-SVar:DBTransform:DB$ SetState | ConditionDefined$ Remembered | ConditionPresent$ Card.ExiledWithSource | ConditionCompare$ GE3 | Defined$ Self | Mode$ Transform
-T:Mode$ ChangesZone | ValidCard$ Card.Self | Origin$ Battlefield | Destination$ Any | Execute$ DBCleanup | Static$ True | Secondary$ True | TriggerDescription$ Forget all remembered cards.
-SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
+A:AB$ ChangeZone | Cost$ 3 W B | ValidTgts$ Creature | Origin$ Battlefield | Destination$ Exile| SubAbility$ DBTransform | SpellDescription$ Exile target creature. Then if there are three or more cards exiled with CARDNAME, transform it.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/p/profane_procession_tomb_of_the_dusk_rose.txt:4]: spacing after Destination

**Comment by tool4ever** [comment]
For the TODO one option might be to store exiledWith as CardTraitBase - but that would require more support in GameState first.

I think this could be merged soon, then I'll have a better base for cleaning up more scripts

**Comment by Hanmac** [comment]
> For the TODO one option might be to store exiledWith as CardTraitBase - but that would require more support in GameState first.
> 
> I think this could be merged soon, then I'll have a better base for cleaning up more scripts

i don't know remember if my latest work for exileWith is online on github now

but the idea was to store the info not in the Exiled Card but in the one that exiled them, but use the CardTraitBase as Param

Then the Card can handle which of the CardTraits form a Linked Connection

**Comment by tool4ever** [comment]
Yea we still have it for future reference!
https://github.com/Card-Forge/forge/tree/exiledWithRework

**Pull request merged**